numba/np/ufunc/dufunc.py

```diff
@@ -321,3 +323,33 @@ def _install_cg(self, targetctx=None):
 
 
 array_analysis.MAP_TYPES.append(DUFunc)
+
+
+@intrinsic
+def intr_reduce(typcontext, ft, xt, axist):
+    # Return signature of array being returned
+    sig = types.Array(types.intp, 1, 'C')(ft, xt, axist)
+    ft = sig.args[0]
+
+    def codegen(context, builder, signature, args):
+        f_ir, x_ir, axis_ir = args
+
+        fnty = lc.Type.function(
+            x_ir.type,
+            [f_ir.type, x_ir.type, axis_ir.type]
+        )
+
+        fn = cgutils.get_or_insert_function(
+            builder.module, fnty,  "dufunc_reduce_direct"
```

Comment:
The current issue I was facing is that this call cannot detect the implemented `dufunc_reduce_direct` method in C code. Is it a `type` issue ?

Reply:
You need to add the symbol - try using `llvmlite.binding.add_symbol('dufunc_reduce_direct', <addr of dufunc_reduce_direct>)`